packages/core/test/test-util/test-services/v2/test-service/TestEntity.ts

```diff
@@ -126,7 +126,7 @@ export class TestEntity extends EntityV2 implements TestEntityType {
   /**
    * One-to-one navigation property to the [[TestEntitySingleLink]] entity.
    */
-  toSingleLink!: TestEntitySingleLink;
+  toSingleLink?: TestEntitySingleLink | null;
```

Comment:
Question: you could also removed the `?` and make it `TestEntitySingleLink | null | undefined`? I wonder how we do it mainly in our code?

Reply:
There is a difference between those two approaches and we use it depending on which makes more sense. If it is optional you can omit the parameter, if it can be undefined you can set it to undefined, but not omit it. Here I think optional is better.

Reply:
For `toSingleLink`, typically, there are three cases:
- an object, as the one-to-one link.
- `null`, because one would like to explicitly set the value to `null`.
- `undefined`, if the value is not set. So it's default value.

If I can choose, I would rather remove the `?`, since as a user, I don't see the use case for setting a `undefined`.
The major reason, why I have to add `?` is to make the compiler happy. Otherwise there will be an error `TS2564: Property 'toSingleLink' has no initializer and is not definitely assigned in the constructor.`

Since `undefined` is not a valid use case, I would not add it to the type and only mark it as optional with `?`.
Here is a [style guide](https://github.com/SAP/cloud-sdk-js/blob/main/STYLEGUIDE.md#use-optional-parameters-for-optional-values)

Reply:
But given the user sets this value to undefined, what is the difference between undefined and null?
I believe what it is (and should be) is that with undefined if you send it towards a system it is ignored, but with null it is explicitly "deleted". We should also mention that in the changelog.